plugins/woocommerce/src/Caches/OrderCountCacheService.php

```diff
@@ -31,6 +33,38 @@ final public function init() {
 		add_action( 'woocommerce_order_status_changed', array( $this, 'update_on_order_status_changed' ), 10, 4 );
 		add_action( 'woocommerce_before_trash_order', array( $this, 'update_on_order_trashed' ), 10, 2 );
 		add_action( 'woocommerce_before_delete_order', array( $this, 'update_on_order_deleted' ), 10, 2 );
+		add_action( 'woocommerce_order_count', array( $this, 'refresh_cache' ) );
```

Comment:
It could just be me, but this took significantly longer on the consumer side to set up and test these hooks.  While the API in this PR is significantly simpler than https://github.com/woocommerce/woocommerce/pull/55746, this is the trade-off we need to consider.

Reply:
While I prefer the more light weight implementation here and think it is more direct, I understand the concern.  

In the end, I would accept either approach.